compiler/rustc_hir_analysis/src/hir_ty_lowering/dyn_compatibility.rs

```diff
@@ -204,9 +204,6 @@ impl<'tcx> dyn HirTyLowerer<'tcx> + '_ {
                                 .filter(|item| item.is_type())
```

Comment:
it's not rly `needed_associated_types` anymore, is it. It's `ordered_associated_types` or whatever. Pls rename the variable and add a comment explaining the approach of

- collect all associated types in a stable order by walking over the dyn trait with identity substitutions
- then use that to order user annotations
- then with elaborated super bounds, not replying user annotations